esphome/components/tcs34725/tcs34725.cpp

```diff
@@ -180,27 +177,21 @@ void TCS34725Component::calculate_temperature_and_lux_(uint16_t r, uint16_t g, u
 }
 
 void TCS34725Component::update() {
-  uint16_t raw_c;
-  uint16_t raw_r;
-  uint16_t raw_g;
-  uint16_t raw_b;
+  uint8_t data[8];  // Buffer to hold the 8 bytes (2 bytes for each of the 4 channels)
 
-  if (this->read_data_register_(TCS34725_REGISTER_CDATAL, raw_c) != i2c::ERROR_OK) {
-    this->status_set_warning();
-    return;
-  }
-  if (this->read_data_register_(TCS34725_REGISTER_RDATAL, raw_r) != i2c::ERROR_OK) {
-    this->status_set_warning();
-    return;
-  }
-  if (this->read_data_register_(TCS34725_REGISTER_GDATAL, raw_g) != i2c::ERROR_OK) {
-    this->status_set_warning();
-    return;
-  }
-  if (this->read_data_register_(TCS34725_REGISTER_BDATAL, raw_b) != i2c::ERROR_OK) {
+  // Perform burst
+  if (this->read_register(TCS34725_REGISTER_CRGBDATAL, data, 8) != i2c::ERROR_OK) {
     this->status_set_warning();
+    ESP_LOGW(TAG, "Error reading TCS34725 sensor data");
     return;
   }
+
+  // Extract the data
+  uint16_t raw_c = data[0] | (data[1] << 8);  // Clear channel
+  uint16_t raw_r = data[2] | (data[3] << 8);  // Red channel
+  uint16_t raw_g = data[4] | (data[5] << 8);  // Green channel
+  uint16_t raw_b = data[6] | (data[7] << 8);  // Blue channel
```

Comment:
We have functions in `helpers.h` that do this with the correct casting. It will require `#include "esphome/core/helpers.h"` at the top too
```suggestion
  uint16_t raw_c = encode_uint16(data[1], data[0]);  // Clear channel
  uint16_t raw_r = encode_uint16(data[3], data[2]);  // Red channel
  uint16_t raw_g = encode_uint16(data[5], data[4]);  // Green channel
  uint16_t raw_b = encode_uint16(data[7], data[6]);  // Blue channel
```